tests/tests/test_mtls.py

```diff
@@ -191,6 +259,40 @@ def test_mtls_enterprise(self, setup_ent_mtls, algorithm):
         out = device.run("mender -show-artifact").strip()
         assert out == "mtls-artifact"
 
+    @MenderTesting.fast
+    @pytest.mark.parametrize("algorithm", ["rsa"])
+    def test_mtls_enterprise_hsm(self, setup_ent_mtls, algorithm):
+        # set up the mTLS test environment and return the device
+        device = self.common_test_mtls_enterprise(setup_ent_mtls, algorithm, True)
+
+        # prepare a test artifact
+        with tempfile.NamedTemporaryFile() as tf:
+            artifact = make_script_artifact(
+                "mtls-artifact", conftest.machine_name, tf.name
+            )
+            deploy.upload_image(artifact)
+
+        # deploy the update to the device
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        assert len(devices) == 1
+        deployment_id = deploy.trigger_deployment(
+            "mtls-test", artifact_name="mtls-artifact", devices=devices,
+        )
+
+        # now just wait for the update to succeed
+        deploy.check_expected_statistics(deployment_id, "success", 1)
+        deploy.check_expected_status("finished", deployment_id)
+
+        # verify the update was actually installed on the device
+        out = device.run(
+            "export SOFTHSM2_CONF=/softhsm/softhsm2.conf; mender -show-artifact"
+        ).strip()
+        assert out == "mtls-artifact"
+
+        self.hsm_cleanup(device)
```

Comment:
This one should be called in a `try/finally` fashion, as the test might abort before reaching here.

Reply:
done.